layers/+spacemacs/spacemacs-defaults/funcs.el

```diff
@@ -479,15 +479,61 @@ If the universal prefix argument is used then will the windows too."
             (if dedicated "no longer " "")
             (buffer-name))))
 
-;; http://camdez.com/blog/2013/11/14/emacs-show-buffer-file-name/
-(defun spacemacs/show-and-copy-buffer-filename ()
-  "Show and copy the full path to the current file in the minibuffer."
+
+;; Copy file path
+(defun spacemacs/copy-directory-path ()
+  "Copy and show the directory path of the current buffer.
+If the buffer is not visiting a file, use the `list-buffers-directory'
+variable as a fallback to display the directory, useful in buffers like the
+ones created by `magit' and `dired'."
+  (interactive)
+  (if-let (directory-name (or
+                            (if-let (file-name (buffer-file-name))
+                              (file-name-directory file-name))
+                            list-buffers-directory))
+    (message "%s" (kill-new (file-truename directory-name)))
+    (message "WARNING: Current buffer does not have a directory!")))
+
+(defun spacemacs/copy-file-path ()
+  "Copy and show the file path of the current buffer."
+  (interactive)
+  (if-let (file-name (buffer-file-name))
+    (message "%s" (kill-new (file-truename file-name)))
+    (message "WARNING: Current buffer is not attached to a file!")))
+
+(defun spacemacs/copy-file-path-with-line ()
+  "Copy and show the file path of the current buffer, including line number."
   (interactive)
-  ;; list-buffers-directory is the variable set in dired buffers
-  (let ((file-name (or (buffer-file-name) list-buffers-directory)))
-    (if file-name
-        (message (kill-new file-name))
-      (error "Buffer not visiting a file"))))
+  (if-let (file-name (buffer-file-name))
+    (let* ((current-file-path (file-truename file-name))
+            (current-line-number (number-to-string (line-number-at-pos)))
+            (complete-message (concat current-file-path ":" current-line-number)))
+      (message "%s" (kill-new complete-message)))
+    (message "WARNING: Current buffer is not attached to a file!")))
+
+(defun spacemacs/copy-file-path-with-line-column ()
+  "Copy and show the file path of the current buffer, including line and column number.
+
+This function respects the value of the `column-number-indicator-zero-based'
+variable."
+  (interactive)
+  (if-let (file-name (buffer-file-name))
+    (let* ((current-file-path (file-truename file-name))
+            (current-line-number (number-to-string (line-number-at-pos)))
+            (current-column-numnber
```

Comment:
Typo in `current-column-numnber`—but you were consistent about it, so it does not make the code incorrect.